src/custom/components/CowSubsidyModal/utils.ts

```diff
@@ -0,0 +1,29 @@
+import { BigNumber } from 'bignumber.js'
+import { COW_SUBSIDY_DATA } from './constants'
+
+export function getDiscountFromBalance(balance: BigNumber) {
```

Comment:
overkill? idk probably

Reply:
Overkill indeed!

What about making sure the threshold is sorted, then only compare the balance to thresholds?

If you must you could store the list of threshold BNs, but you are traversing the array too many times to little gain IMO.

What about something like:

```ts
// assuming COW_SUBSIDY_DATA is sorted
// alternatively could create a sorted Array<(threshold, tier)>
for (const [threshold, tier] of COW_SUBSIDY_DATA) {
  const tBN = new BigNumber(threshold)
  if (balance.lt(tBN)) {
    return tier
  }
}
return COW_SUBSIDY_DATA[last item]
```

Reply:
changed in #2558